### DISCUSSION TITLE
SNMP Pull specific Interface

### DISCUSSION THREAD

**Discussion by remchuk** [prometheus/snmp_exporter#1110] [Q&A]
Hey there!

I want to give a variable of a device + interface, lets say "SwitchA + Ethernet1/34" to the snmp and it will give me only his metrics.
I want to do this for lets say several devices. 

Is there a way to pull the metrics of specific interface?
I know that dynamic filtering exists, but it works with regex rather then a specific interface lets say.

**Comment by SuperQ** [comment]
You have to specify exactly which OIDs as get, not walk.

The problem is that SNMP doesn't really have the kind of server-side request filtering. It's a very simple protocol.
